### ISSUE TITLE
[Feature] Performer Minigames

### ISSUE THREAD

**Issue by sleetx** [stashapp/stash#3823]
Here's an idea for a new "Performer Minigames" feature. You select the gender of performer to use, and then one of the following games:

- **Versus mode:** Pick the winner between two random performers. Their overall winning percentage is displayed. 

- **FMK:** Three performers are displayed and you have to choose which ones to Fuck, Marry, Kill. Each performer's overall F, M, and K percentages are displayed.

- **Guess Who:** One performer image is displayed, and you have to guess their name between four options.

After each game, you are given the option to play again with new random performers. A mockup of the "Versus mode" game is shown below:
![image](https://github.com/stashapp/stash/assets/46305211/a5780dd8-4a8d-48de-ae7d-f16de267e424)

**Comment by QxxxGit** [comment]
I was just thinking about this the other day. I think it's out of the scope for Stash but games should definitely be made by the community.

I'd like to see some ideas for games that can help improve manual metadata entry like ratings and such. I think those would be really nice.

**Comment by echo6ix** [comment]
I like where this idea is going, but it might get more interest if there was more emphasis on _versus mode_, and if this was framed less as minigames and more as a [Pairwise Comparison](https://en.wikipedia.org/wiki/Pairwise_comparison) tool. Pairwise comparison is a legitimate process to derive a hierarchy of preference based rankings.

## TLDR on pairwise comparison
If someone reading this does not know what pairwise comparison is (colloquial called Versus Mode here, I think) it's basically presented exactly like OPs concept image.

1. It starts with a random pair of items, for this example it will be a pair of performer images.
2. The user selects which presented item they prefer.
3. Based on the user's choice, the rating of the two items are updated using one of the mainly available[^g] pairwise algorithms.
4. Repeat steps 2 and 3 with additional pairs of items, randomly selected each time.
5. Continue serving items to select until all pairs of items have been compared (but continue process indefinitely)

As the process gets closer to completion (where completion means all available pairs are rated against each another), the hierarchal ranking accrues more accuracy. For Stash purposes the process could continue indefinitely, there is no issue of rating pairs against one another multiple times to revise the rankings.

## How is this different than the rating stars?
Stash already offers a rating system, but rating stars are also the most unsophisticated method, and it is also much less engaging for Stashnerds that want a more objective rating system and crave some diverse engagement.

Rating stars rate an item on a numerical scale independent of the other items being rated. Pairwise comparison method directly compares items in the set against each other to derive a rating relative to other items in the set. This approach eliminates some of the subjectivity involved in rating stars and it provides a more structured and robust ranking system.

- Sophisticated way to rate sets of items (performers, studios, images, movies, tags, etc, it does not need to be limited to performers)
- Interactive and engaging (appeals to people who like the subject matter and more engagement with their Stash)
- Gives users a better idea of what items they actually do prefer and the types of rankings that could be developed are endless
  - People have been craving badges. Badges can be derived from the different types of pairwise comparison rankings
- Could potentially integrate into Stash in the sense that you could sort items based on their pairwise rank in the grid view modes


## Does it belong in Stash? Should it be a plugin?
This is actually not the first time a "versus mode" has been proposed. It seems like many years ago now the user Fonzie on Discord seemed to have fleshed out a "versus mode", but obviously nothing came of it and he's long since been inactive. IIRC the reception to his concept was pretty good.

While this may seem outside the scope of Stash, especially how the OP has framed it, technically this is just a more sophisticated way of rating items of preference. It just so happens that it can be viewed as a game when you're rating content of interest.

For people who only horde and organize their content, this may seem outside the scope of how they use Stash. But for people who enjoy engaging with their content beyond being an archivist, a _versus mode_ would probably be a fun way to develop neat analytics.

I don't know how this would work on the technical side of things. Maybe this is more feasible as a plugin that integrates into Stash. I dont know.

[^g]:Considering the utility and legitimacy of Pairwise Comparison, there are probably [many Pairwise Comparison libraries right here on Github](https://github.com/search?q=pairwise&type=repositories) that can be incorporated into this concept. This tool would not need to be created from scratch.

**Comment by AdultSun** [comment]
I agree with Qxxx on this one too, could be fun and/or useful but outside of the scope for Stash. I feel like any games like this would have to be a plugin or external tool that hooks into Stash or Stash-Box's API somehow instead.

**Comment by WithoutPants** [comment]
Agreed. This is out of scope of stash, but could definitely be a plugin or outside tool that hooks into the api. Closing this an not planned, but I don't mean to close discussion.
